### ISSUE TITLE
Visible error in Clustering samples>parallel

### ISSUE THREAD

**Issue by phisanti** [bigomics/omicsplayground#564]
When filtering the samples by gene and then selecting a gene family, the all the boards show an error. I think this is likely because the selected gene is not present in the filter. Thus, it could be introduced some kind of pre-filter to the choices in the Gene Family. Another option, could be to set a "Compute" button and block the calculation when the user selects a gene that is not within the Gene Family option.

![image](https://github.com/bigomics/omicsplayground/assets/33370667/354257ec-cb29-4e19-9f8b-a83ecc58a392)

**Comment by phisanti** [comment]
I got this one, but I might need some feedback. So the bug emerges from `omicsplayground\components\board.clustering\R\clustering_server.R` line 77. The original code looks like this:


```r
    ## update choices upon change of level
    shiny::observe({
      shiny::req(pgx$families, pgx$gsetX)
      shiny::req(input$hm_level)
      choices <- names(pgx$families)
      if (input$hm_level == "geneset") {
        nk <- sapply(playdata::COLLECTIONS, function(k) sum(k %in% rownames(pgx$gsetX)))
        choices <- names(playdata::COLLECTIONS)[nk >= 5]
      }
      choices <- c("<custom>", "<contrast>", choices)
      choices <- sort(unique(choices))
      shiny::updateSelectInput(session, "hm_features", choices = choices)
    })
```

I then changed the observe to observeEvent() and filter according to the selected gene, if "Split samples by" gene is on. Here is the code:

```r
    ## update choices upon change of level
    shiny::observeEvent(input$hm_splitvar, {
      shiny::req(pgx$families, pgx$gsetX)
      shiny::req(input$hm_level)
      choices <- names(pgx$families)
      if (input$hm_level == "gene") {
        nk <- sapply(playdata::COLLECTIONS, function(k) sum(k %in% rownames(pgx$gsetX)))
        choices <- names(playdata::COLLECTIONS)[nk >= 5]
        subset_choices <- sapply(choices, function(x) any(x == input$hm_splitvar))
        choices <- names(subset_choices)
      }
      choices <- c("<custom>", "<contrast>", choices)
      choices <- sort(unique(choices))
      shiny::updateSelectInput(session, "hm_features", choices = choices)
    })

```

With the new code, the Gene Family only presents options where the chosen gene is present, thus, solving the error. However, I noticed that  `if (input$hm_level == "geneset")` depends on an option that is hidden in advanced options, so I am no sure if the change still keeps the intended behaviour. Any help will be appreciated before commit. @ivokwee @ncullen93

**Comment by ESCRI11** [comment]
Closed by #579
